### DISCUSSION TITLE
using php functions block event loop ?

### DISCUSSION THREAD

**Discussion by foremtehan** [swoole/swoole-src#5085] [General]
I'm a laravel octane user. I want to know if I use php functions like `sleep` or `file_get_contents` etc..., do these functions block the whole event loop, even if I use them inside the swoole `go` function  or `Octane::concurrently`? If the answer is yes, how can I find the equivalent of every blocking PHP function in Swoole?

**Comment by NathanFreeman** [comment]
Yes, Laravel octane does not support coroutines currently and it will block whole event if use php functions like sleep or file_get_contents in the laravel octane.

You can try to use swoft, easyswoole, hyperf, imi framework if you want to use coroutine.
